regolith/helpers/a_todohelper.py

```diff
@@ -0,0 +1,126 @@
+"""Helper for adding a projectum to the projecta collection.
+
+   Projecta are small bite-sized project quanta that typically will result in
+   one manuscript.
+"""
+import datetime as dt
+import dateutil.parser as date_parser
+from dateutil.relativedelta import relativedelta
+import sys
+
+from regolith.helpers.basehelper import DbHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import (
+    all_docs_from_collection,
+    get_pi_id,
+)
+
+TARGET_COLL = "todolist"
+Importance = [0, 1, 2, 3, 4, 5]
+
+
+def subparser(subpi):
+    subpi.add_argument("description", help="the description of the to_do task",
+                       default=None)
+    subpi.add_argument("-i", "--id", help="ID of the to_do list's owner. The default id is saved in user.json. ")
+    subpi.add_argument("-d", "--due_date",
+                       help="Proposed due date for the to_do task."
+                       )
+    subpi.add_argument("-p", "--importance",
+                       help=f"The importance of this task from {Importance}. Default is 0 (unimportant).",
+                       default=0
+                       )
+    subpi.add_argument("-e", "--estimated_time",
+                       help="The estimated time that the task will take. The unit is minute. Default is 60 ",
+                       default = 60
+                       )
+    return subpi
+
+
+class TodoAdderHelper(DbHelperBase):
+    """Helper for adding a projectum to the projecta collection.
+
+       Projecta are small bite-sized project quanta that typically will result in
+       one manuscript.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "a_todo"
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+
+        rc.coll = f"{TARGET_COLL}"
+        rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def db_updater(self):
+        rc = self.rc
+        if not rc.id:
+            try:
+                rc.id = rc.default_user_id
+            except AttributeError:
+                raise RuntimeError(
+                    "Please set default_user_id and user_first_name in '~/.config/regolith/user.json',\
+                     or you need to input your id in the command line")
+        now = dt.date.today()
+        if not rc.due_date:
```

Comment:
I wonder if it is better to make due date required (but easy to type.... so select from tomorrow, two days, one week, one month or allow an ISO date if there is a more precise idea)